tools/logger/convert.c

```diff
@@ -284,6 +287,16 @@ static void process_params(struct proc_ldc_entry *pe,
 			p += 2;
 		}
 	}
+
+	/* ensure no NULLs in format strings */
+	for (j = 0; j < i; j++) {
+		if (!pe->params[j]) {
+			pe->params[j] = (uintptr_t)missing;
+			/* ensure object is no free()'ed */
+			pe->subst_mask &= ~(1 << j);
+		}
+	}
```

Comment:
This feels sketchy. Do we even support _strings_ in the parameters in the first place? I thought only stuff like %d (%x, %o) and %p are supported in parameters. I feel like this is a correctness issue. But then you would also have to scan the format string to "ban" the %s specifier which is more complicated.

Reply:
strings not supported today and unlikely to be since this is all free with the Zephyr logger.

Reply:
@paulstelian97 we do support strings. Here's example print for params when subst_mask is set:

```
params[0] 0x9E19E370 (subst_mas clear)
params[1] 'dma-trace-task <2b972272-c5b1-4b7e-926f-0fc5cb4c4690>' (subst_mask set)
```

And matching log output:
```
[         131.770828] (         131.770828) c0 ll-schedule        ./schedule/ll_schedule.c:513  INFO task add 0x9e19e370 dma-trace-task <2b972272-c5b1-4b7e-926f-0fc5cb4c4690>
```

It's still sketchy. Let me see if I can do a simpler change.

Reply:
I'm confused how strings can be supported because sof-logger uses dictionary logging and I'm not aware of any support for strings in there. Zephyr logging I think is not based on dictionary logging by default?

Reply:
@paulstelian97 I think Zephyr now has the string dictionary support, initially all the strings were baked into the Zephyr image (sorry, I don't know the Zephyr internals here). xtos config does not support ```%s``` strings as we cant do stuff like ```__func__``` in our trace today, only strings are in the trace body via the dictionary.

Reply:
Okay, I want to learn about the Zephyr logger and see if (after NXP migrates to fully Zephyr) we can use that. Right now we still have to use sof-logger due to still keeping the xtos builds (Zephyr isn't fully stable for us, there's a good few quirks left to fix)

Reply:
@paulstelian97 @lgirdwood There's still code to handle "%s" but this is not used by any code in SOF FW and sof-logger emits a warning if one is seen. But we have two indirect usages of "%s", FW traces with "%pUx" and "%pQ" are converted to "%s" and the argument passed from FW (in raw log stream) is converter to a string that is passed as argument to (f)printf. And these steps can hit an alloc failure, or just missing conversion, leading to NULL being used. 

But right, "plain %s" usage where you'd have %s used directly in FW code, is not there.